src/app/home/home.page.ts

```diff
@@ -124,14 +124,10 @@ export class HomePage {
   }
 
   filterCountries(searchBarValue: string | undefined | null): void {
-    const query = searchBarValue?.toLowerCase().trim();
-    if (query && query !== '') {
-      this.countries = this.isHomePage ? this.baseCountries.filter(country => country.name.toLowerCase().indexOf(query) > -1)
-        : this.favourites.filter(country => country.name.toLowerCase().indexOf(query) > -1);
-      this.groupCountries();
-    } else {
-      this.updateView();
-    }
+    const query = (searchBarValue ?? '').toLowerCase().trim();    
```

Comment:
1- use Nullish Coalescing Operator (??): Ensures searchBarValue is an empty string if null or undefined, simplifying subsequent operations.

2-  Combines the filtering logic into a single chain operation, improving readability without sacrificing clarity.

3- Direct Assigns the filtered result directly to this.countries, reducing unnecessary variable assignments